crates/proof-of-sql/src/sql/proof_plans/union_exec.rs

```diff
@@ -58,13 +62,16 @@ where
     ) -> Result<TableEvaluation<S>, ProofError> {
         let gamma = builder.try_consume_post_result_challenge()?;
         let beta = builder.try_consume_post_result_challenge()?;
+        let mut num_mle_evaluations = None;
         let c_star_evals = self
             .inputs
             .iter()
             .map(|input| -> Result<_, ProofError> {
                 let table_evaluation =
                     input.verifier_evaluate(builder, accessor, None, chi_eval_map, params)?;
-                let c_fold_eval = gamma * fold_vals(beta, table_evaluation.column_evals());
+                let column_evals = table_evaluation.column_evals();
+                num_mle_evaluations = num_mle_evaluations.or(Some(column_evals.len()));
```

Comment:
OK this is weird. If we want to have the number of columns in the union we may as well make it a field and let `UnionExec::try_new` figure it out. No need for a mutable variable that is really only set once.